### PR TITLE
Add support for data-at-rest encryption

### PR DESCRIPTION
**Pull request by LaPetiteSouris**
To solve: https://github.com/liftbridge-io/liftbridge/issues/217

Required: https://github.com/liftbridge-io/liftbridge-api/pull/52

Prerequisite of: https://github.com/liftbridge-io/liftbridge/pull/338

### CODE REVIEW

File: v2/client.go [hunk 1]
```diff
@@ -182,6 +182,9 @@ type StreamOptions struct {
 	// OptimisticConcurrencyControl controls the activation of optimistic concurrency control
 	// of the stream
 	OptimisticConcurrencyControl *bool
+
+	// EncryptionDataAtRest controls the activation of encryption data-at-rest on server side
+	EncryptionDataAtRest *bool
```
Comment (tylertreat) [inline v2/client.go:187]: Let's just call this `Encryption`.

File: v2/client.go [hunk 2]
```diff
@@ -420,6 +426,15 @@ func OptimisticConcurrencyControl(val bool) StreamOption {
 	}
 }
 
+// EncryptionDataAtRest sets the value of EncryptionDataAtRest, which enables
+// encryption of data at rest on the server side for the given partition
+func EncryptionDataAtRest(val bool) StreamOption {
```
Comment (tylertreat) [inline v2/client.go:431]: Same thing here.

**Review by tylertreat — COMMENTED** [review]
lgtm. Just a minor nitpick about naming.

File: v2/client.go (hunk 1 above)
Comment (LaPetiteSouris) [reply v2/client.go]: fixed

File: v2/client.go (hunk 2 above)
Comment (LaPetiteSouris) [reply v2/client.go]: fixed

**Comment by LaPetiteSouris** [comment]
The test/build will pass once https://github.com/liftbridge-io/liftbridge-api/pull/52 is merged and `go.mod` is updated

**Comment by LaPetiteSouris** [comment]
`lifbridge-api` updated. I guess it's OK for my side .

**Pull request merged**